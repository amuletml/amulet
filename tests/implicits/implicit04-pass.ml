type foo 'a = Foo of 'a

let foo_int = Foo 1

let bar : forall 'a. foo 'a => 'a -> unit = fun _ x -> ()
let foo : forall 'a. foo 'a => 'a -> unit = fun _ x -> bar x
