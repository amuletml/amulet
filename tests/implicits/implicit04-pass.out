foo : type -> type
Foo : {'a : type}. 'a -> foo 'a
foo_int : foo int
bar : {'a : type}. foo 'a => 'a -> unit
foo : {'a : type}. foo 'a => 'a -> unit
